README.md

```diff
@@ -86,27 +80,126 @@ $POSTGRES_PASSWORD  # postgres's user's password
+60 examples, 0 failures
+[root@9eea4caf7740 code]# rake
+Migrating to latest
+/usr/local/bin/ruby app.rb
+== Sinatra (v1.4.7) has taken the stage on 4567 for development with backup from Thin
+Thin web server (v1.7.0 codename Dunder Mifflin)
+Maximum connections set to 1024
+Listening on localhost:4567, CTRL+C to stop
+```
+
+You may be prompted to `bundle install`.
+
+## Development Without `docker`
+
+If you do not want to install `docker` and `docker-copmose` and instead want to
+run the service natively you can try installing the following packages:
+
+- `sqlite-devel`
+- `postgresql`
+- `postgresql-devel`
+- `ruby-2.3.0` and `gem`
+- `gem install bundle`
+
+Unfortunately we are not able to help with getting `postgresql` up and running
+on your system, but there's plenty of tutorials that *can* help with that.
+
+If you do have problems installing your system, try reading the
+`dockerfiles/app` file to see how the docker environment is created (it's
+basically a bash script). If you're still having problems setting up your
+environment please feel free to make an issue and we'll help out.
+
+### Setting up the Environment
+
+You'll need to add a few environment variables to your computer. I suggest
+adding the following the lines to your `.bashrc`:
+
+```
+export POSTGRES_PASSWORD=changeme   # your postgres password
+export CACHE_FILE=/tmp/iam-cache    # Where you want to save the cache file to
+export POSTGRES_HOST=127.0.0.1      # Whatever your postgres host ip address is
+```
+
+and reload your shell with `source ~/.bashrc`.
+
+### Running tests
+
+```
+$ rake spec
+[... tests running ...]
+Finished in 0.1669 seconds (files took 0.36978 seconds to load)
+60 examples, 0 failures
+$ rake
+Migrating to latest
+/usr/local/bin/ruby app.rb
+== Sinatra (v1.4.7) has taken the stage on 4567 for development with backup from Thin
+Thin web server (v1.7.0 codename Dunder Mifflin)
+Maximum connections set to 1024
+Listening on localhost:4567, CTRL+C to stop
+```
+
 ### Troubleshooting
-When you try to run the app with `rake spec` or `ruby app.rb` and receive errors like:
+
+**Problem:** When you try to run the app with `rake spec` or `ruby app.rb` and
+receive errors like:
+
 ```
 Could not find parser-2.3.0.7 in any of the sources
 Run `bundle install` to install missing gems.
 ```
 
-This represents that your docker image has gem dependencies that are out of date.
+This represents that your docker image has gem dependencies that are out of
+date.
 
 **To fix:** Run `docker-compose build --no-cache`. This will rebuild the
 container and rerun the command `bundle install` to refresh all the gems
+
+**Problem:** ``docker`` or ``docker-compose`` does not work, or you get an
+error along the lines of
+
+```
+$ docker-compose run dev bash
+Could not find file <some file>
+[root@9eea4caf7740 code]#
+```
+
+or you might get something like `too many symbolic links`.
+
+Just `exit` the docker environment and run
+
+```
+$ sudo systemctl restart docker
```

Comment:
I think that you should either make this inline code (like the previous problem's solution) or make the other problem's solution a code block like this one.